packages/datadog-instrumentations/src/multer.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+
+const shimmer = require('../../datadog-shimmer')
+const { channel, addHook, AsyncResource } = require('./helpers/instrument')
+
+const multerReadCh = channel('datadog:multer:read:finish')
+
+function publishRequestBodyAndNext (req, res, next) {
+  return shimmer.wrapFunction(next, next => function () {
+    if (multerReadCh.hasSubscribers && req) {
+      const abortController = new AbortController()
+      const body = req.body
+
+      multerReadCh.publish({ req, res, body, abortController })
```

Comment:
Since body is already part of req, is there a reason we need to pass it separately?
(We do the same thing also in body-parser)

Reply:
A while ago I asked the question: "Who should be responsible of finding the data ? the instrumentation publisher ? or the instrumentation subscriber ?"
Everyone agreed that it was better if it was the responsibility of the instrumentation publisher (ie: this file) to "find" the body property.
That allows us to write framework independent subscribers.
Makes sense to you now ?

Reply:
we could get it from the `req`, yes. But I guess it's a way to decouple them. Making unnecessary for the listeners to know that the body is a property of the `req`

Reply:
Example: we could have 3 different body parser libraries, that have different kind of APIs etc, but the instrumentation point all publish with the same standardized arguments, thus allowing us to only have one subscriber for all 3 libraries.
The instrumentation acts as a "translator", it is the part of the code with the most knowledge about the context, and thus should be responsible for abstracting that in a standard signature

Reply:
Yes I got your point, thanks for clarification